### PR TITLE
Cancel payment immediately on USER_CANCELLED (bug 1129449)

### PR DESCRIPTION
**Pull request by muffinresearch**

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -81,6 +81,10 @@ spa.get('/mozpay/', function (req, res) {
     // Setup a view that sets a example of an error like webpay would
     // but this time withn no error code attr.
     context.startView = 'payment-failed';
+  } else if (req.originalUrl === '/mozpay/provider/reference/user-cancelled') {
+    // Setup a view that sets a example of an error like webpay would.
+    context.startView = 'payment-failed';
+    context.errorCode = 'USER_CANCELLED';
```
Comment (kumar303) [inline server/index.js]: is Webpay fully updated to set the error code and context? I'm not sure it is (for Bango at least): https://github.com/mozilla/webpay/blob/master/webpay/bango/views.py#L128

Comment (kumar303) [inline server/index.js]: hmm, not sure if I'm looking at the right view there but that doesn't appear to be using SPA even

**Comment by kumar303** [comment]
r+wc

This change looks good but Webpay might need an update

File: server/index.js (hunk 1 above)
Comment (muffinresearch) [inline server/index.js]: Yeah it uses system_error which leads to this: https://github.com/mozilla/webpay/blob/master/webpay/base/utils.py#L77

**Comment by muffinresearch** [comment]
I'll test this out tomorrow. Should be good though.

**Pull request merged**